### PR TITLE
LIME-1128 - Feature-tests - API Test Framework Deployment (Promotion Tests for HMRC KBV)

### PR DESCRIPTION
**Pull request by ChrisBates1**
### What changed

Initial POC Typescript API Test Framework for HMRC KBV CRI
Endpoints under Test include (fetchquestions, question, answer)
Includes Dockerfile and bash script to execute tests

### Why did it change

To provide a basis for future Pre-Merge FeatureTests to be created

### Issue tracking

- [LIME-1128](https://govukverify.atlassian.net/browse/LIME-1128)

### Other considerations

Action fixes added for Fraud, DL and Passport have been included in this commit for ipv-cri-kbv-hmrc-ap https://govukverify.atlassian.net/browse/LIME-843

README.md file added to feature-tests dir with details on params required to be set to run locally and docker build and execute commands

[LIME-1128]: https://govukverify.atlassian.net/browse/LIME-1128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature-tests/apiEndpoints/endpoints.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default class EndPoints {
+  public static PRIVATE_API_GATEWAY_URL =
+    process.env.TEST_ENVIRONMENT === 'dev'
+      ? `https://${process.env.SAM_STACK_NAME}.execute-api.eu-west-2.amazonaws.com/${process.env.TEST_ENVIRONMENT}`
```
Comment (gds-jbentham) [inline feature-tests/apiEndpoints/endpoints.ts:4]: I dont think we need to differentiate for dev here
`https://${process.env.CFN_PrivateApiGatewayId}.execute-api.eu-west-2.amazonaws.com/${process.env.TEST_ENVIRONMENT}`
Should work for all

File: feature-tests/results/report.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gds-jbentham) [inline feature-tests/results/report.xml:1]: This file can be removed

File: feature-tests/run-tests.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+cp /jest.config.ts . 2>/dev/null || :
+cp /jest-cucumber-config.js . 2>/dev/null || :
+cp -R /tests ./tests 2>/dev/null || :
+cp -R /apiEndpoints ./apiEndpoints 2>/dev/null || :
+cp -R /node_modules ./node_modules 2>/dev/null || :
+cp -R /utils ./utils 2>/dev/null || :
+
+# run tests and save the exit code
+declare test_run_result
+export tagFilter=@regression
```
Comment (gds-jbentham) [inline feature-tests/run-tests.sh:16]: Can we set this from a parameter so we can change it in different environments
  export tagFilter =$(aws ssm get-parameter --name "/tests/${STACK_NAME}/TestTag" | jq -r ".Parameter.Value")

Comment (ChrisBates1) [reply feature-tests/run-tests.sh]: Export filter for Test Tag update on run-tests.sh file

File: feature-tests/apiEndpoints/endpoints.ts (hunk 1 above)
Comment (ChrisBates1) [reply feature-tests/apiEndpoints/endpoints.ts]: This method has been updated to use the .env value for the PRIVATE_API_GATEWAY_URL
removed === dev statement

**Review by ElliotMurphyGDS — DISMISSED** [review]

**Review by smsgds — APPROVED** [review]

**Pull request merged**